File: cpp/sprint5/H/code.cpp
#ifndef REMOTE_JUDGE
struct Node {  
  int value;  
  const Node* left = nullptr;  
  const Node* right = nullptr;
  Node(int value, Node* left, Node* right) : value(value), left(left), right(right) {}
};  
#endif

#ifdef REMOTE_JUDGE
#include "solution_tree.h"
#endif
#include <cmath>
#include <iostream>
#include <cassert>

using namespace std;


int Solution(const Node* root) {
    // Your code
    // “ヽ(´▽｀)ノ”
} 

#ifndef REMOTE_JUDGE
void test() {
    Node node1({2, nullptr, nullptr});
    Node node2({1, nullptr, nullptr});
    Node node3({3, &node1, &node2});
    Node node4({2, nullptr, nullptr});
    Node node5({1, &node4, &node3});
    assert(Solution(&node5) == 275);
}

int main() {
  test();
}
#endif

File: go/sprint5/H/code.go
package main


// <template>
type Node struct {
	value  int
	left   *Node
	right  *Node
}
// <template>

func Solution(root *Node) int {
    // Your code
    // “ヽ(´▽｀)ノ”
}

func test() {
	node1 := Node{2, nil, nil}
    node2 := Node{1, nil, nil}
    node3 := Node{3, &node1, &node2}
    node4 := Node{2, nil, nil}
    node5 := Node{1, &node4, &node3}
	if Solution(&node5) != 275 {
		panic("WA")
	}
}
